pydantic/networks.py

```diff
@@ -351,6 +393,114 @@ class PostgresDsn(AnyUrl):
     }
     user_required = True
 
+    __slots__ = ('hosts',)
+
+    def __init__(self, *args, hosts: Optional[List['HostParts']] = None, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.hosts = hosts
+
+    @classmethod
```

Comment:
surely we can make some fairly small changes to `AnyUrl` to reduce the amount we need to duplicate here?

Reply:
Makes sense, DRY rule 💪